.github/workflows/static-checking.yml

```diff
@@ -27,20 +27,30 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - name: Install Requirements
+        if: ${{ matrix.python-version != '3.12' }}
         run: |
           python -m pip install --upgrade pip
           python -m pip install poetry
           poetry config virtualenvs.create false --local
           poetry install --all-extras -vvv
+      - name: Install Requirements
+        if: ${{ matrix.python-version == '3.12' }}
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install poetry
+          poetry config virtualenvs.create false --local
+          poetry install --extras "redshift mysql postgres sqlserver oracle gremlin sparql opencypher opensearch openpyxl progressbar deltalake geopandas" -vvv
       - name: Black style
         run: black --check .
       - name: ruff check
         run: |
           ruff . --ignore "PL" --ignore "D"
           ruff awswrangler
       - name: mypy check
+        if: ${{ matrix.python-version != '3.12' }}
```

Comment:
Does mypy not support 3.12 yet?

Reply:
It does. The issue is that without Ray and Modin, we get a bunch of errors from the `awswrangler/distributed/*` subpackage. But I'll add a comment here to clarify this, since we want to remove this condition as soon as Ray begins to support Python 3.12.

Reply:
Why did we decide to expand the pyversions to 3.11 and 3.12 in the first place? We only had 3.8 to start with...

Perhaps we can park the typing required changes in a separate branch for now until ray supports py3.12?

Reply:
I think I originally expanded this workflow to include multiple Python versions just to make sure I wasn't breaking things as I was upgrading dependencies. But you're right, we don't need additional Python versions here. I'll change this to only include Python 3.8 again.